[TVM][CUDA] NVIDIA GPU Int8 Support

nishi-t opened the pull request:
This PR adds int8 support for nvidia gpus

tqchen commented:
@vinx13 since you are working on related stuff, can you take a look?

vinx13 commented on src/codegen/codegen_cuda.cc:
```diff
@@ -83,7 +87,8 @@ void CodeGenCUDA::PrintType(Type t, std::ostream& os) {  // NOLINT(*)
       // directly 4 8 bit int in integer.
-      os << "int"; return;
+      enable_int8_ = true;
+      os << "char4"; return;
```
@tqchen do we need to support other lanes size? e.g. int4 if lanes == 16

tqchen replied:
This, this would actually be very helpful, @vinx13 can you elaborate what are possible things we need to support and value translation rules in here?

tqchen commented:
@nishi-t can you act on @vinx13 's comment? being able to perform full vector load will be very helpful to get the full perf. 

Specifically, we want to be able to load 4 words from memory at a time, unfortunately there is no char16 struct so we have two solutions:

- Use int2 and int4 for these vectors, while not support arithmatics for these as @vinx13 suggested.
- Try to deifne a char16 and support arithmetics

int2 and int4 might be a easier path for now if we only need save/load and get nvidia's native support

nishi-t commented:
@vinx13 Thank you for the suggestion. Could you please review again?
@tqchen For now, I addressed the comment by first solution. And, I'll try second solution seperately in another PR, because it may take a little while. Is this okay?

vinx13 commented:
looks good from my side

nishi-t commented:
@tqchen Yes, I'll address it.
@vinx13 Thank you.

nishi-t commented:
@tqchen I addressed. Please review again.

tqchen commented:
Thanks @nishi-t @vinx13 , this is now merged
